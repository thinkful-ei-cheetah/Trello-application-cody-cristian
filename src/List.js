import React from 'react';
import Cards from './Card.js' 
import './List.css';
function List(props){
    return(
        <section className="List">
          <header className="List-header">
            {props.header}
          </header>
          <div className="List-cards">
          {props.cards.map((card)=>
            <Cards key={card.id} title={card.title} content={card.content} />
    )}
            
          </div>
        </section>
        
       
    )
}
export default List