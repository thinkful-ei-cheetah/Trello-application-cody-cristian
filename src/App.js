import React, { Component } from 'react';
import List from './List.js' 
import './App.css';

class App extends Component {
  render() {
    return (
      <main >
        <List className='List-header'>
          <h2>First list</h2>
        </List>
      </main>
    );
  }
}

export default App;
