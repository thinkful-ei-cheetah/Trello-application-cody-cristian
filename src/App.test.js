import React from 'react';
import ReactDOM from 'react-dom';
import Cards from './Card.js'
import List from './List'
import STORE from './Store.js'
import App from './App';
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Cards/>,div)
});
it('renders the UI as expected', () => {
  const tree = renderer
   .create(<Cards />)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });
// list.js
it('renders', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <List key={STORE.lists.id} header='test' cards={STORE.lists.map((id) => <Cards title={id.title} content={id.content} /> )} />,div)
  });
  it('renders the UI as expected', () => {
    const tree = renderer
     .create(<List key={STORE.lists.id} header='test' cards={STORE.lists.map((id) => <Cards title={id.title} content={id.content} />)} />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });