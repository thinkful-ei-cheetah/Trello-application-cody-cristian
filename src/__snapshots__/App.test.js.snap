// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    test
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3 />
      <p />
    </div>
    <div
      className="Card"
    >
      <h3 />
      <p />
    </div>
    <div
      className="Card"
    >
      <h3 />
      <p />
    </div>
    <div
      className="Card"
    >
      <h3 />
      <p />
    </div>
  </div>
</section>
`;
